File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    if (!root) return null;
    let max = -Infinity;
    const DFS = (node, depth) => {
        if (!node.left && !node.right) {
            max = Math.max(max, depth);    
        }
        node.left && DFS(node.left, depth + 1);
        node.right && DFS(node.right, depth + 1);
    }
    DFS(root, 1)
    return max;
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode], depth=0) -> int:
        if not root:
            return depth
        
        left = self.maxDepth(root.left, depth + 1) 
        right = self.maxDepth(root.right, depth + 1) 
        return max(left, right)